### PR TITLE
ASoC: rt722-sdca: Add RT722 SDCA driver

### PR DESCRIPTION
**Pull request by jack-cy-yu**
This is the initial codec driver for rt722-sdca.

### CODE REVIEW

File: sound/soc/codecs/rt722-sdca-sdw.c [hunk 1]
```diff
@@ -0,0 +1,505 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// rt722-sdca-sdw.c -- rt722 SDCA ALSA SoC audio driver
+//
+// Copyright(c) 2022 Realtek Semiconductor Corp.
```
Comment (RanderWang) [inline sound/soc/codecs/rt722-sdca-sdw.c]: it is 2023 in rt722-sdca-sda.h

File: sound/soc/codecs/rt722-sdca.h [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+#define RT722_EAPD_HIGH				0x2
+#define RT722_EAPD_LOW				0x0
+
+/* Buffer address for HID */
+#define RT722_BUF_ADDR_HID1			0x44030000
+#define RT722_BUF_ADDR_HID2			0x44030020
+
+/* RT722 SDCA Control - function number */
+#define FUNC_NUM_JACK_CODEC 0x01
```
Comment (RanderWang) [inline sound/soc/codecs/rt722-sdca.h]: tab align like above ?

File: sound/soc/codecs/rt722-sdca.c [hunk 3]
```diff
@@ -0,0 +1,1549 @@
...
+#include <sound/tlv.h>
+
+#include "rt722-sdca.h"
+
+int rt722_sdca_index_write(struct rt722_sdca_priv *rt722,
+		unsigned int nid, unsigned int reg, unsigned int value)
+{
+	int ret;
+	struct regmap *regmap = rt722->mbq_regmap;
+	unsigned int addr = (nid << 20) | reg;
```
Comment (RanderWang) [inline sound/soc/codecs/rt722-sdca.c:32]: move "int ret;" here "

**Review by RanderWang — COMMENTED** [review]
LGTM

**Review by RanderWang — APPROVED** [review]

**Comment by plbossart** [comment]
this is now reviewed upstream, let's continue the threads on alsa-devel. Thanks @jack-cy-yu

**Pull request closed without merging**